js/symb/leafletMap.js

```diff
@@ -0,0 +1,201 @@
+const DEFAULT_DRAW_OPTIONS = {
+   polyline: false,
+   control: true,
+   circlemarker: false,
+   marker: false,
+   multiDraw: false,
+   drawColor: { color: '#000', opacity: 0.85, fillOpacity: 0.55 }
+};
+
+const DEFAULT_MAP_OPTIONS = {
+   center: [43.64701, -79.39425],
+   zoom: 15,
+};
+
+const DEFAULT_SHAPE_OPTIONS = {
+   color: '#000',
+   opacity: 0.85,
+   fillOpacity: 0.55
+};
+
+class LeafletMap {
+   /* To Hold Reference to Leaflet Map */
+   mapLayer;
+
+   /* Active_Shape Type
+    * type: String represenation of shape 
+    * layer: leaflet layer
+    * center: { lat: float, lng float }
+    */
+   activeShape;
+
+   /* Reference Leaflet Feature Group for all drawn items*/
+   drawLayer;
+
+   constructor(map_id, map_options=DEFAULT_MAP_OPTIONS) {
+      this.mapLayer = L.map(map_id, map_options);
+
+      const terrainLayer = L.tileLayer('https://{s}.google.com/vt?lyrs=p&x={x}&y={y}&z={z}', {
```

Comment:
Sorry. Where are `s`, `x`, `y`, and `z` actually coming from?

Reply:
They aren't coming from anywhere in the code it is a definition that is apart of the link so it is just a string